src/NUnitFramework/framework/Internal/Execution/SimpleWorkItem.cs

```diff
@@ -140,6 +140,13 @@ private TestCommand MakeTestCommand()
                 foreach (var attr in method.GetCustomAttributes<IApplyToContext>(true))
                     command = new ApplyChangesToContextCommand(command, attr);
 
+                // Add construct and optionally dispose command in case of instance per test case.
+                if (parentFixture?.LifeCycle == LifeCycle.InstancePerTestCase)
+                {
+                    command = new ConstructFixtureCommand(command);
+                    if (typeof(IDisposable).IsAssignableFrom(Test.TypeInfo.Type))
+                        command = new DisposeFixtureCommand(command);
+                }
```

Comment:
Also good! I'd just prefer you had a new command rather than modifying behavior of `ConstructFixtureCommand`. Obviously, we can't have two `ConstructFixtureCommand`s, so some renaming may be in order.

At some point, I'd like to do a pass through the commands to make them more atomic. I can see other places where what I call "policy logic" has slipped into the commands.

Reply:
agreed, naming things is one of the hardest things in programming ;)
perhaps ConstructInstancePerTestCaseCommand ? i'm not sure if i'm allowed to change the original ConstructFixtureCommand to ConstructSingleInstanceCommand

Reply:
I agree with @CharliePoole here. I think the fact that you are trying to make the `ConstructFixtureCommand` do two things is cascading out into other complexity. Walking up from the test to the suite isn't appropriate there because the class wasn't intended to work on individual tests.

I'm not sure that you need to rename `ConstructFixtureCommand` as that has always been our default behavior. Maybe a new command `ConstructFixturePerTestCaseCommand`?

Reply:
i agree @rprouse  - `ConstructFixturePerTestCaseCommand` sounds good, all that's left is the when, see above since im having trouble trying to control it via `Context` (not applied soon enough in the command chain to control via `MakeOneTimeSetUpCommand` )